nextcord/application_command.py

```diff
@@ -3276,3 +3293,177 @@ def unpack_annotation(
+            Tuple[EllipsisType, int],
+        ],
+    ) -> Type[int]:
+        ...
+
+    @overload
+    def __class_getitem__(
+        cls,
+        value: Union[
+            float,
+            Tuple[float, float],
+            Tuple[float, EllipsisType],
+            Tuple[EllipsisType, float],
+        ],
+    ) -> Type[float]:
+        ...
+
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            float,
+            Tuple[int, int],
+            Tuple[float, float],
+            Tuple[Union[int, float], EllipsisType],
+            Tuple[EllipsisType, Union[int, float]],
+        ],
+    ) -> Type[Union[int, float]]:
+        class Inner(Range, OptionConverter):
+            def __init__(self):
+                super().__init__(option_type=type(self.min or self.max))
+
+            def modify(self, option: SlashCommandOption):
+                if self.min:
+                    if option.min_value is None:
+                        option.min_value = self.min
+                if self.max:
+                    if option.max_value is None:
+                        option.max_value = self.max
+
+        if isinstance(value, tuple):
+            if len(value) != 2:
+                raise ValueError("Range can only take 1-2 arguments")
+
+            min = value[0]
+            max = value[1]
+        else:
+            min = None
+            max = value
+
+        if min is None or isinstance(min, EllipsisType):
+            Inner.min = None
+        elif isinstance(min, (int, float)):
+            Inner.min = min
+        else:
+            raise TypeError("Range min must be int or float.")
+
+        if isinstance(max, EllipsisType):
+            Inner.max = None
+        elif isinstance(max, (int, float)):
+            Inner.max = max
+        else:
+            raise TypeError("Range max must be int or float.")
+
+        if Inner.min is None and Inner.max is None:
+            raise TypeError("At least one of min or max must be set.")
+
+        return Inner
+
+
+class String(int):
+    """An annotation helper for defining slash command `min_length` and `max_length` parameters.
+
+    .. versionadded:: 2.2
+
+    .. container:: operations
+
+        .. describe:: String[x, y]
+
+            Creates a range of string length from ``x`` to ``y``.
+
+        .. describe:: String[x] | String[..., x]
+
+            Create a range of string length up to ``x``.
+
+        .. describe:: String[x, ...]
+
+            Create a range of string length from ``x``.
+    """
+
+    min: ClassVar[Optional[int]]
+    max: ClassVar[Optional[int]]
+
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            Tuple[int, int],
+            Tuple[int, EllipsisType],
+            Tuple[EllipsisType, int],
+        ],
+    ) -> Type[int]:
+        class Inner(String, OptionConverter):
+            def __init__(self):
+                super().__init__(option_type=str)
```

Comment:
Currently a NotImplemented error is raised when running a command with Range

```suggestion
        class Inner(String, OptionConverter):
            def __init__(self):
                super().__init__(option_type=str)
                
            async def convert(self, interaction: Interaction, value: Any) -> Any:
                return value
```